Mage/src/main/java/mage/game/stack/Spell.java

```diff
@@ -1057,27 +1058,38 @@ public void checkForCountersToAdd(Permanent permanent, Ability source, Game game
     }
 
     @Override
-    public StackObject createCopyOnStack(Game game, Ability source, UUID newControllerId, boolean chooseNewTargets) {
-        return createCopyOnStack(game, source, newControllerId, chooseNewTargets, 1);
+    public void createCopyOnStack(Game game, Ability source, UUID newControllerId, boolean chooseNewTargets) {
+        createCopyOnStack(game, source, newControllerId, chooseNewTargets, 1);
     }
 
     @Override
-    public StackObject createCopyOnStack(Game game, Ability source, UUID newControllerId, boolean chooseNewTargets, int amount) {
+    public void createCopyOnStack(Game game, Ability source, UUID newControllerId, boolean chooseNewTargets, int amount) {
+        createCopyOnStack(game, source, newControllerId, chooseNewTargets, amount, null);
+    }
+
+    @Override
+    public void createCopyOnStack(Game game, Ability source, UUID newControllerId, boolean chooseNewTargets, int amount, SpellCopyApplier applier) {
         Spell spellCopy = null;
         GameEvent gameEvent = new CopyStackObjectEvent(source, this, newControllerId, amount);
         if (game.replaceEvent(gameEvent)) {
-            return null;
+            return;
         }
+        // TODO: add a way to choose order of items on the stack
```

Comment:
@JayDi85 do you have any ideas on how to do this? Would we be able to use the same GUI as for placing multiple triggers on the stack?

Reply:
Ideally there are must be a choose order dialog with target names (like choice but with up/down button to change sort order) or like mtga mouse moving.

But yes, you can use any existing dialogs. Also it can be simple choice dialog like:
* choose copied spells to put on stack (spell text here);
* key items list to choose (target names or none -- you don't need rules here cause it's same);
* chosen spell goes to stack and user start to choose another one. 
* If nothing chosen/cancel/disconnected then put all last spells at once. AI (`player.isComputer`) put it on copying order without choice.

Reply:
BTW, if there are possible combo with many copies then it can have additional choice item like "put as it" with random uuid key -- so users can select it to "skip" choice dialog to the end.

Reply:
Alright, how would I go about implementing that? I don't really know the frontend stuff.

Reply:
You don't need frontend. Just use choice dialog. Search code for `// choice dialog` or `setKeyChoices` to find a `ChoiceImpl` usage example.

Reply:
oh ok, thanks

Reply:
Alright, I implemented the dialog but the tests fail now because they can't handle the copy dialog. Is there a way to have the computer player choose cancel when it has a choice?

Reply:
If you need it for `chooseTarget` then try to add `TestPlayer.TARGET_SKIP` in the addTarget test command to cancel targeting.

If you need it for for choices then find `TestPlayer.java` and `public boolean choose(Outcome outcome, Choice choice, Game game) {` , add skip code and use `TestPlayer.CHOICE_SKIP` in setChoice test command:
``` java
  // skip choices
  if (choices.get(0).equals(CHOICE_SKIP)) {
      choices.remove(0);
      return true;
  }
```
![shot_210312_025033](https://user-images.githubusercontent.com/8344157/110865881-be7a4700-82dd-11eb-8244-eea655bd9ca9.png)

Reply:
It's about unit tests and a Human logic testing. If you need a different logic for computer then use `player.isComputer` in the code and add some "AI hint" to make correct choices.

Reply:
Setting the choice outcome to `AIDontUseIt` seems to work fine for having the AI cancel but adding the skip you gave there doesn't work as it gets caught in here:
![image](https://user-images.githubusercontent.com/29759218/110866801-b2699780-8293-11eb-95ae-87d6087ca020.png)

Reply:
I see a code call:
player.choose(Outcome.AIDontUseIt, choice, game);

So it must be used by TestPlayer from my comment above.

If your skip command "eating" too early then you must setup correct test commands order. It's easy to disable all commands, enable it one by one and run it to see a strict mode fails -- so you can see a real chooses order and can setup the answers.

Reply:
Oh I see, it's not choosing the starting player automatically.

Reply:
Nope, starting player is chooses automaticity in the tests somthere and it's playerA all the time. You must not setup it at all. Something wrong. I'll debug and fix it later.

Reply:
ok yeah that seemed off to me

Reply:
Look at the end of `TestPlayer.choose` method from your screenshot:
``` java
public boolean choose(Outcome outcome, Target target, UUID sourceId, Game game, Map<String, Serializable> options)
```

And try to move `// ignore player select` to the start:
* At the end code:
``` java
this.chooseStrictModeFailed("choice", game, getInfo(game.getObject(sourceId)) + ";\n" + getInfo(target));
return computerPlayer.choose(outcome, target, sourceId, game, options);
```
![shot_210312_084910](https://user-images.githubusercontent.com/8344157/110893666-da96dc00-830f-11eb-96b3-aeb8c8c6fc25.png)

* At the start code before choices:
``` java
// ignore player select
if (target.getMessage().equals("Select a starting player")) {
    return computerPlayer.choose(outcome, target, sourceId, game, options);
}
```
![shot_210312_084844](https://user-images.githubusercontent.com/8344157/110893631-c8b53900-830f-11eb-9953-980a362a9975.png)

I think that changes can fix a starting player skip bug and you don't need to setup names in unit tests. If it does then PR can be merged. All other code looks ok.

Reply:
I applied the changes you suggested and had to make one small change to `PlayerLeftGameRange1Test.TestCaptiveAudienceGoesToExile`, is that alright?

Reply:
Yes , all fine.